src/ess.jl

```diff
@@ -328,10 +336,12 @@ function ess_rhat(
         sum_pₜ = pₜ
 
         k = 2
-        while k < maxlag
+        while true
             # compute subsequent autocorrelation of all chains
             # by combining estimates of each chain
             ρ_even = 1 - inv_var₊ * (W - mean_autocov(k, esscache))
+            # stop summation if the next even lag would exceed maxlag. this ρ_odd is unused.
+            k < maxlag - 1 || break
```

Comment:
I wonder if it would be clearer to handle this last iteration separately after the `while`/`for` loop?